File: ios/PolkaChat/Rust/UI.swift
//
//  UI.swift
//  PolkaChat
//
//  Created by Daniel Leping on 07/03/2023.
//

import Foundation

import CPolkaChat

public class UI {
    private let errorModel: ErrorModel
    
    init(errorModel: ErrorModel) {
        self.errorModel = errorModel
    }
    
    func presentError(message: String) {
        Task {
            await self.errorModel.presentError(message: message)
        }
    }
    
    func asRust() -> SUI {
        SUI(ui: self)
    }
}

File: common/src/ios/ui/ui.rs
use crate::ui::UIProtocol;
use crate::error::Result as PCResult;

use super::sui::SUI;

pub (crate) struct UI {
    swift: SUI
}

impl UI {
    pub (crate) fn new(sui: SUI) -> Self {
        UI {
            swift: sui
        }
    }
}

impl UIProtocol for UI {
    fn present_error(&self, message: &str) -> PCResult<()> {
        self.swift.present_error(message)
    }
}